torch/ao/quantization/fx/prepare.py

```diff
@@ -508,6 +534,40 @@ def _get_arg_target_dtype_as_output(
     # TODO: should support is_dynamic here as well
     return output_act_dtype
 
+
+# TODO: merge with _get_arg_target_dtype_as_output
+def _get_arg_target_is_dynamic_as_output(
+    arg: Node,
+    named_modules: Dict[str, torch.nn.Module],
+) -> Optional[Union[torch.dtype, type]]:
+    """ Get the target output activation dtype for
+    the argument in the original graph, skipping inserted observers
+    We are assuming that the observers are inserted correctly, and the dtype for
+    argument in quantized graph will match what is specified by the qconfig
+    """
+    assert isinstance(arg, Node)
+    # Custom module LSTM output is a tuple that we broke down into the internal nodes in order
+    # to insert DeQuantStubs (see `_insert_dequant_stubs_for_custom_module_lstm_output`).
+    # Since we modified the graph in this case, we must trace back from the args through
+    # the specific nodes we added in order to reach the original LSTM node. Otherwise, we would
+    # not be able to accurately detect whether this node is a consumer of custom module LSTM.
```

Comment:
Why is this the case? And if this is because we inserted observers, then should be do node annotation so that you dont have to traverse the modified graph?

Reply:
you mean the LSTM related comment? this is for fx graph mode quantization

Reply:
I think I understand now. The lstm part is unrelated to annotation and already has modified graph with dequant stub.